src/libraries/analytics/analyticsEventTracking.ts

```diff
@@ -541,7 +545,10 @@ const AnalyticsEventTracking = (() => {
   });
 
   return {
-    eventTrack: (event: BetterSocialEventTracking, additionalData?: object): Promise<void> => {
+    eventTrack: (
+      event: BetterSocialEventTracking | string,
+      additionalData?: object
+    ): Promise<void> => {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `eventTrack` now accepts a string as an event parameter. This could potentially introduce a risk of arbitrary event tracking if the input is not properly validated or sanitized. Consider adding some form of validation or sanitization to ensure that only expected events are tracked.
```diff
    eventTrack: (
      event: BetterSocialEventTracking | string,
      additionalData?: object
    ): Promise<void> => {
+     if (typeof event === 'string' && !BetterSocialEventTracking[event]) {
+       console.error('Invalid event');
+       return Promise.resolve();
+     }
      if (!event) {
        console.error('Event must be defined');
        return Promise.resolve();
```
This will check if the event is a string and if it's not part of the `BetterSocialEventTracking` enum, it will log an error and exit the function.


<!-- This is an auto-generated comment by OSS CodeRabbit -->